share/p2p/discovery/options.go

```diff
@@ -14,45 +21,56 @@ type Parameters struct {
 	// Set -1 to disable.
 	// NOTE: only full and bridge can advertise themselves.
 	AdvertiseInterval time.Duration
+	// Tag is used as rondezvous point for discovery service
+	Tag string
+}
+
+// options is the set of options that can be configured for the Discovery module
+type options struct {
+	// onUpdatedPeers will be called on peer set changes
+	onUpdatedPeers OnUpdatedPeers
 }
 
 // Option is a function that configures Discovery Parameters
-type Option func(*Parameters)
+type Option func(*options)
 
 // DefaultParameters returns the default Parameters' configuration values
 // for the Discovery module
-func DefaultParameters() Parameters {
-	return Parameters{
+func DefaultParameters() *Parameters {
+	return &Parameters{
 		PeersLimit:        5,
 		AdvertiseInterval: time.Hour,
+		//TODO: remove fullNodesTag default value once multiple tags are supported
+		Tag: fullNodesTag,
 	}
 }
 
 // Validate validates the values in Parameters
 func (p *Parameters) Validate() error {
-	if p.AdvertiseInterval <= 0 {
+	if p.Tag == "" {
```

Comment:
Because it was incorrect. `AdvertiseInterval` == -1 disables it. We didn't caught it because config validation was called outside of the constructor, which happen to be before options were applied. So it was a bug